src/Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore/DbContextEditorServices.cs

```diff
@@ -171,6 +187,76 @@ public EditSyntaxTreeResult EditStartupForNewContext(ModelType startUp, string d
             };
         }
 
+        private IPropertySymbol TryGetIConfigurationRootProperty(ITypeSymbol startup)
+        {
+            var propertySymbols = startup.GetMembers()
+                .Select(m => m as IPropertySymbol)
+                .Where(s => s != null);
+
+            foreach (var pSymbol in propertySymbols)
+            {
+                var namedType = pSymbol.Type as INamedTypeSymbol; //When can this go wrong?
+                if (namedType != null &&
+                    namedType.ContainingAssembly.Name == "Microsoft.Extensions.Configuration.Abstractions" &&
+                    namedType.ContainingNamespace.ToDisplayString() == "Microsoft.Extensions.Configuration" &&
+                    namedType.Name == "IConfigurationRoot") // What happens if the type is referenced in full in code??
+                {
+                    return pSymbol;
+                }
+            }
+
+            return null;
+        }
+
+        // Internal for unit tests.
+        internal void AddConnectionString(string connectionStringName, string dataBaseName)
+        {
+            var appSettingsFile = Path.Combine(_applicationInfo.ApplicationBasePath, "appsettings.json");
+            JObject content;
+            bool writeContent = false;
+
+            if (!_fileSystem.FileExists(appSettingsFile))
+            {
+                content = new JObject();
+                writeContent = true;
+            }
+            else
+            {
+                content = JObject.Parse(_fileSystem.ReadAllText(appSettingsFile));
+            }
+
+            string dataNodeName = "Data";
+            string connectionStringNodeName = "ConnectionString";
+
+            if (content[dataNodeName] == null)
+            {
+                writeContent = true;
+                content[dataNodeName] = new JObject();
+            }
+
+            if (content[dataNodeName][connectionStringName] == null)
+            {
+                writeContent = true;
+                content[dataNodeName][connectionStringName] = new JObject();
+            }
+
+            if (content[dataNodeName][connectionStringName][connectionStringNodeName] == null)
```

Comment:
Instead doing this you might want to have a model/class which then would be automatically converted to json (see how runtime read/write to project.lock.json)